tide_site.install

```diff
@@ -239,3 +240,28 @@ function tide_site_update_8010() {
   $config_storage = \Drupal::service('config.storage');
   $config_storage->write($view, $source->read($view));
 }
+
+/**
+ * Update approver and site_admin roles permissions.
+ *
+ * 1. approver and site_admin cannot delete sites terms.
+ * 2. revoke `administer taxonomy` permission.
+ * 3. assign `access taxonomy overview` permission.
+ */
+function tide_site_update_8011() {
+  $roles = ['approver', 'site_admin'];
+  $taxonomy_permissions = new TaxonomyPermissions(\Drupal::entityManager());
+  foreach ($roles as $role_id) {
```

Comment:
@vincent-gao I think this is fine in an update hook but if this were a function that was to be reused it would be preferable to make use of the Drupal APIs. For example
[Role::getPermissions](https://api.drupal.org/api/drupal/core%21modules%21user%21src%21Entity%21Role.php/function/Role%3A%3AgetPermissions/8.7.x)
[user_role_permissions](https://api.drupal.org/api/drupal/core%21modules%21user%21user.module/function/user_role_permissions/8.7.x)
[TaxonomyPermissions::buildPermissions](https://api.drupal.org/api/drupal/core%21modules%21taxonomy%21src%21TaxonomyPermissions.php/function/TaxonomyPermissions%3A%3AbuildPermissions/8.7.x) - this would be helpful to set the vocabulary  and remove the need for the check.

Reply:
Thanks @GROwen